src/VisualStudio/Core/Def/Implementation/InlineRename/InlineRenameUndoManager.cs

```diff
@@ -218,27 +218,26 @@ private IOleUndoManager GetUndoManager(ITextBuffer subjectBuffer)
 
             private IEnumerable<IOleUndoUnit> GetUndoUnits(IOleUndoManager undoManager)
             {
-
                 // Unfortunately, EnumUndoable returns the units in oldest-first order.
-                undoManager.EnumUndoable(out var undoUnitEnumerator);
-                if (undoUnitEnumerator == null)
-                {
-                    yield break;
-                }
+                IEnumOleUndoUnits undoUnitEnumerator = null;
+                var hr = ErrorHandler.CallWithCOMConvention(() => undoManager.EnumUndoable(out undoUnitEnumerator));
 
-                const int BatchSize = 100;
-                while (true)
+                if (ErrorHandler.Succeeded(hr) && undoUnitEnumerator != null)
```

Comment:
📝 The subexpression `undoUnitEnumerator != null` will always be true.

Reply:
Is this guaranteed? Could you point me to the implementation please?

Reply:
@genlu It's a standard COM convention. If you return a successful status code, the return pointers are expected to be valid. Several cases in native code do not check for null after seeing `S_OK`.